src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -107,17 +104,16 @@ class Snapshot:
     """
 
     def __init__(self) -> None:
-        self._realization_states: Dict[
-            str,
-            Dict[str, Union[bool, datetime, str, Dict[str, "ForwardModel"]]],
-        ] = defaultdict(dict)
-        """A shallow dictionary of realization states. The key is a string with
-        realization number, pointing to a dict with keys active (bool),
-        start_time (datetime), end_time (datetime) and status (str)."""
-
-        self._forward_model_states: DefaultDict[Tuple[str, str], ForwardModel] = (
-            defaultdict(ForwardModel)  # type: ignore
-        )
+        self._realization_states: DefaultDict[
+            RealId,
+            RealizationSnapshot,
+        ] = defaultdict(RealizationSnapshot)  # type: ignore
```

Comment:
check if the `type: ignore` can be dropped.

Reply:
can it?

Reply:
it can't. mypy has problems with defaultdict taking typeddicts as argument. you can write defaultdict(lambda: RealizationSnapshot()), but then pyright will give you warnings that you should write defaultdict(RealizationSnapshot)...